### ISSUE TITLE
[DV Regression Results] Upload to OpenTitan doc server

### ISSUE THREAD

**Issue by sriyerg** [lowRISC/opentitan#1322]
I would like to be able to upload the results to the OT web server whenever regressions are run. For starters, it will just be a markdown / html file (sample attached). Eventually, it will also include the tool generated coverage report (several html files). The goal is to have the tool directly copy over the results to the web server once every night (skip GitHub). It is sufficient to maintain the history of the last 7 days. 

Can we maintain something like this: 
```console
docs.opentitan.org/hw/ip/<ip>/dv/results 
 |-> latest -> soft link to the latest
 |-> 2019.01.10 -> latest
 |-> 2019.01.09 -> previous  
 ...  
 |-> 2019.01.04 -> oldest 
 |-> v1 -> snapshot of the results when v1 milestone was achieved 
 |-> v2 -> --"-- 
 |-> v3 -> --"-- 
```
At the end of the regression, the oldest directory is deleted and a new one is created. The 'latest' soft link is updated. The reason for maintaining this soft link is to have the 'V' status in the [dashboard](https://docs.opentitan.org/doc/project/hw_dashboard/) always point to the latest results. 

Please provide some help in setting this up. 

Thanks 
Sri

**Comment by sriyerg** [comment]
Sample regr_results.md file:

```md
# UART Regression Results
  Run on Friday January 10 2020 01:36:06PM 

## Test Results
| Milestone   | Name               | Tests             | Results   |
|-------------|--------------------|-------------------|-----------|
| V1          | csr                | uart_csr_hw_reset | 1/1       |
| V1          |                    | TOTAL             | 1/1       |
| V2          | outstanding_access | uart_csr_hw_reset | 1/1       |
| V2          | partial_access     | uart_csr_hw_reset | 1/1       |
| V2          |                    | TOTAL             | 2/2       |
|             |                    | TOTAL             | 3/3       |
```

**Comment by gkelly** [comment]
The easiest thing to do would be add a service account with access to the docs serving bucket.  Where do the reports get generated?  I'd like to make sure that a compromise of that serving key doesn't end up with docs.opentitan.org defaced or serving malware, so it might be preferable to add a new subdomain like reports.opentitan.org and give that process access to that bucket only.

**Comment by gkelly** [comment]
To follow up here: I set up [reports.opentitan.org](https://reports.opentitan.org) and have it serving from a separate bucket.  I created a service account and provided the key to Sri.  We discussed on a call that if it's more than one machine running this process, or if it's going to be multiple developers doing it interactively, then it makes sense to switch to authenticating/acting as the developer identity instead of a service account.
